src/types/error.ts

```diff
@@ -0,0 +1,10 @@
+export interface Error {
```

Comment:
You may want to take a look into https://angular.io/api/common/http/HttpErrorResponse. If you have, I'm wondering why was it insufficient to be used?